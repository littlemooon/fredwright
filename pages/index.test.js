// @flow
/* eslint-env jest */

import { mount } from 'enzyme'
import React from 'react'
import renderer from 'react-test-renderer'

import Index from './index.js'

describe('With Enzyme', () => {
  it('Index shows title', () => {
    const x = mount(<Index url={{ pathname: '/' }} />)
    expect(x).toIncludeText(`HOME`)
    expect(x).toIncludeText(`oh g'mornin`)
  })
})

describe('With Snapshot Testing', () => {
  it('Index renders', () => {
    const component = renderer.create(<Index url={{ pathname: '/' }} />)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
