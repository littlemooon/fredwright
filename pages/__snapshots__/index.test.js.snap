// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`With Snapshot Testing Index renders 1`] = `
<div
  className="jsx-2738761292"
>
  <div
    className="jsx-2738761292"
  >
    <h1
      className="jsx-1117296120"
    >
      HOME
    </h1>
    <div
      className="jsx-3581857328"
    >
      <ul
        className="jsx-1117296120"
      >
        <li
          className="jsx-1117296120"
        >
          <a
            className="jsx-1117296120"
            href="/"
            onClick={[Function]}
          >
            home
          </a>
        </li>
        <li
          className="jsx-1117296120"
        >
          <a
            className="jsx-1117296120"
            href="/about"
            onClick={[Function]}
          >
            about
          </a>
        </li>
      </ul>
    </div>
    <p
      className="jsx-1117296120"
    >
      currently on: /
    </p>
  </div>
  <p>
    oh g'mornin
  </p>
</div>
`;
